pkg/skaffold/initializer/build/resolve.go

```diff
@@ -29,17 +31,25 @@ import (
 // For each image parsed from all k8s manifests, prompt the user for the builder that builds the referenced image
 func (d *defaultBuildInitializer) resolveBuilderImages() error {
 	// If nothing to choose, don't bother prompting
-	if len(d.unresolvedImages) == 0 || len(d.builders) == 0 {
+	if len(d.builders) == 0 {
 		return nil
 	}
 
-	// if we only have 1 image and 1 build config, don't bother prompting
-	if len(d.unresolvedImages) == 1 && len(d.builders) == 1 {
-		d.builderImagePairs = append(d.builderImagePairs, BuilderImagePair{
-			Builder:   d.builders[0],
-			ImageName: d.unresolvedImages[0],
-		})
-		return nil
+	// if there's only one builder config, no need to prompt
+	if len(d.builders) == 1 {
+		if len(d.unresolvedImages) == 0 {
+			// no image was parsed from k8s manifests, so we create an image name
+			d.generatedBuilderImagePairs = append(d.generatedBuilderImagePairs, getGeneratedBuilderPair(d.builders[0]))
+			return nil
+		}
+		// we already have the image, just use it and return
+		if len(d.unresolvedImages) == 1 {
+			d.builderImagePairs = append(d.builderImagePairs, BuilderImagePair{
+				Builder:   d.builders[0],
+				ImageName: d.unresolvedImages[0],
+			})
+			return nil
+		}
```

Comment:
would `len(d.builder) =1` and `(d.unresolvedImaged) >1` never happen?

Reply:
it absolutely would! if `len(d.unresolvedImages()) > 1` we go into `resolveBuilderImagesInteractively()`